cmd/tempo/app/app.go

```diff
@@ -207,14 +207,12 @@ func (t *App) Run() error {
 		// let's find out which module failed
 		for m, s := range serviceMap {
 			if s == service {
-				switch service.FailureCase() {
-				case modules.ErrStopProcess:
-					level.Info(log.Logger).Log("msg", "received stop signal via return error", "module", m, "err", service.FailureCase())
-				case context.Canceled:
-				default:
-					level.Error(log.Logger).Log("msg", "module failed", "module", m, "err", service.FailureCase())
+				err = service.FailureCase()
+				if errors.Is(err, modules.ErrStopProcess) {
+					level.Info(log.Logger).Log("msg", "received stop signal via return error", "module", m, "err", err)
+				} else if err != nil && !errors.Is(err, context.Canceled) {
```

Comment:
this is tough to read. maybe do a:

```
if errors.Is(context.Canceled) {
  continue
}
```

at the top of the loop. any idea why we are ignoring context canceled? also, it looks like the old code logs an error even it's nil?